### PR TITLE
Added a 'Why?' section to the competencies

### LINKED ISSUE #3: We should say *why* these are our competencies
Somewhere, probably in an introductory section, we should briefly say something about why we chose these four categories for our competencies and how they reflect our values as an engineering org.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,6 +176,14 @@ Jump to competencies for a particular level:
 
 ## FAQ
 
+### Why these competencies?
+
+We believe a team should prosper and grow _as a team_: that its success is measured by both the quality and throughput of its output _and_ the personal growth and enjoyment of its individual members.
+Strong technical achievements are only half of that.
+Fostering a collaborative community is also necessary for us to be a great place to work.
+
+We've found that these four competency areas capture much of what's worked well for us.
```
Comment (JoshuaKGoldberg) [inline README.md]: Do ya'll think I should write stuff on why these _specific_ competencies are good and valid?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Codecademy Engineering Competencies
 
-These are the evaluation criteria we use for performance evaluation and leveling on the Codecademy engineering team. The compentencies at each level are broken down into four categories:
+These are the evaluation criteria we use for performance evaluation and leveling on the Codecademy engineering team. Hooray! The compentencies at each level are broken down into four categories:
 
 1. Software Engineering & Design
```
Comment (kipprice) [inline README.md]: I'd rather a bit of "why?" in each one of the listing of competencies

File: README.md (hunk 1 above)
Comment (kipprice) [reply README.md]: Yes, I do

Comment (kipprice) [inline README.md]: I would put your last line first, so it reads at a high level as:
- we have data supporting that these are the right choices
- a development team is more than the sum of its technical chops

File: README.md [hunk 3]
```diff
@@ -176,6 +177,40 @@ Jump to competencies for a particular level:
...
+#### Why Execution & Results?
+
+On top of the core development fundamentals, successful engineers are ones who ship products and are able to reliably do so on time with high quality.
+They need to balance the various tasks they're given and, to an extent, manage their own time.
+Engineers have uniquely deep insight into their work, and often need to drive it forward.
+
+#### Why Collaboration & Communication?
+
+Teams are more than the sum of their parts, and as an engineer, part of your responsibilities are that everyone is given the resources to work their best.
+On one level, we need team members to be active participants in bettering the code and documentation around them.
```
Comment (kipprice) [inline README.md]: These next two points ("on one level...", "on a higher level...") seem more like a plug for community + citizenship; I'd suggest rewording to something along the lines of:

"Strong team members both communicate the current progress and blockers of their own work, in order to keep their peers from getting stuck, and share feedback regularly that shape the code and product to be the best it can be."
